lib/queuePopulator/IngestionPopulator.js

```diff
@@ -114,6 +140,80 @@ class IngestionPopulator {
         this._mProducer.setupProducer(cb);
     }
 
+    /**
+     * Check mongo for new buckets and apply any updates
+     * @param {Function} cb - callback(error)
+     * @return {undefined}
+     */
+    applyUpdates(cb) {
+        this._mongoClient.getIngestionBuckets((err, buckets) => {
+            if (err) {
+                return cb(err);
+            }
+            // Example buckets list:
+            // [
+            //     {
+            //         name: 'my-zenko-bucket',
+            //         ingestion: { Status: 'enabled' },
+            //         locationConstraint: 'my-ring'
+            //     },
+            // ]
+
+            // Example this._locationDetails
+            // {
+            //     'my-ring': {
+            //         accessKey: ...,
+            //         secretKey: ...,
+            //         endpoint: 'http://10.100.1.128:8000',
+            //         locationType: 'scality_s3',
+            //         bucketName: 'source-bucket-1'
+            //     }
+            // }
+
+            const sources = Object.keys(this._provisionedIngestionSources);
+
+            return async.series([
+                done => async.each(buckets, (bucket, next) => {
+                    // zenko bucket name
+                    const key = bucket.name;
+                    if (!sources.includes(key)) {
+                        // Add new ingestion source
+                        const locationDetails =
+                            this._locationDetails[bucket.locationConstraint];
+                        const endpoint = url.parse(locationDetails.endpoint);
+                        // TODO: will change for non-ring sources
+                        const newSource = {
+                            name: bucket.name,
```

Comment:
TODO: add descriptive comments for now
Later, we can rename these key names throughout the code